Detectors/TRD/simulation/src/TrapSimulator.cxx

```diff
@@ -2069,7 +2070,8 @@ void TrapSimulator::fitTracklet()
         }
         bool printoutadcs = false;
         for (int i = 0; i < 21; i++) {
-          if (adchitbp & (1U << i)) {
+          // FIXME it can happen that there is a hit found in a channel where there is no digit provided as input. Is this expected?
+          if ((adchitbp & (1U << i)) && (mADCDigitIndices[i] >= 0)) {
```

Comment:
@bazinski I noticed here that it actually can happen that a hit is found in a channel where there was no digit input. Do you know if this is expected? Because initially I thought this would not happen and therefore I also did not reset the `mADCDigitIndices` array. I noticed this, because for a large number of events the number of labels attached to tracklets differed depending on the order in which the TRAPs are processed. With this workaround it is not the case anymore.